Upstream-status: Partially integrated <https://github.com/sphinx-contrib/httpdomain/pull/110>

TODO: Find a better fix than the HTTPIndex removal, and resubmit to upstream
(probably in Sphinx).

diff --git a/src/sphinxcontrib/httpdomain/__init__.py b/src/sphinxcontrib/httpdomain/__init__.py
index 48bb580..2cf41d7 100644
--- a/src/sphinxcontrib/httpdomain/__init__.py
+++ b/src/sphinxcontrib/httpdomain/__init__.py
@@ -207,6 +207,7 @@ HTTP_STATUS_CODES = {
     304: 'Not Modified',
     305: 'Use Proxy',
     307: 'Temporary Redirect',
+    308: 'Permanent Redirect',
     400: 'Bad Request',
     401: 'Unauthorized',
     402: 'Payment Required',     # unused
@@ -221,7 +222,7 @@ HTTP_STATUS_CODES = {
     411: 'Length Required',
     412: 'Precondition Failed',
     413: 'Request Entity Too Large',
-    414: 'Request URI Too Long',
+    414: 'URI Too Long',
     415: 'Unsupported Media Type',
     416: 'Requested Range Not Satisfiable',
     417: 'Expectation Failed',
@@ -229,6 +230,7 @@ HTTP_STATUS_CODES = {
     422: 'Unprocessable Entity',
     423: 'Locked',
     424: 'Failed Dependency',
+    425: 'Too Early',           # RFC 8470
     426: 'Upgrade Required',
     429: 'Too Many Requests',
     449: 'Retry With',           # proprietary MS extension
@@ -651,7 +653,7 @@ class HTTPDomain(Domain):
         'any': {}
     }
 
-    indices = [HTTPIndex]
+    indices = []
 
     @property
     def routes(self):
